src/client/pythonEnvironments/common/externalDependencies.ts

```diff
@@ -176,6 +177,15 @@ export function getPythonSetting<T>(name: string): T | undefined {
     return vscode.workspace.getConfiguration('python').get(name);
```

Comment:
```suggestion
    const settings = internalServiceContainer.get<IConfigurationService>(IConfigurationService);
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return (settings as any)[name];
```
Instead of this solution, I think it's better if we change this to use Configuration service instead. Configuration service already handles system variables among other things.